lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -47,11 +48,41 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) || $a cmp $b } keys %$clones) {
-        my $url = $clones->{$path};
+        my $url = $clones->{$path} // '';
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
+
+        my $git = OpenQA::Git->new(app => $app, dir => $path);
+        $url //= $git->get_origin_url;
+        my $server_host;
+        if ($url =~ m{^(?:\w+://)?([^/\@:\s]+)(?::\d+)?/}) {
+            # e.g. "https://gitlab.suse.de/qa/foo", "ssh://git.foo.org/repo"
+            $server_host = $1;
+        }
+        elsif ($url =~ m{^(?:[^@]+@)?([^:]+):}) {
+            # e.g. "git@gitlab.suse.de:qa/foo", "user@host:repo"
+            $server_host = $1;
```

Comment:
Can this be in one regex?

Reply:
yes :+1:  now using `Mojo::URL->host` for the http/ssh part and only one regex just for scp-like urls, as previously suggested by tina